plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergTableWithCustomLocation.java

```diff
@@ -52,125 +43,91 @@
         extends AbstractTestQueryFramework
 {
     private FileHiveMetastore metastore;
-
-    private static final String CATALOG = "iceberg";
-    private static final String SCHEMA = "iceberg_schema";
-    private static final String TABLE = "iceberg_table";
-    private static final String TABLE_RENAMED = "iceberg_table_renamed";
+    private File metastoreDir;
 
     @Override
     protected DistributedQueryRunner createQueryRunner()
             throws Exception
     {
-        Session session = testSessionBuilder()
-                .setIdentity(Identity.forUser("hive")
-                        .withRole("hive", new SelectedRole(ROLE, Optional.of("admin")))
-                        .build())
-                .build();
-        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
-
-        File baseDir = queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data").toFile();
-
-        HdfsConfig hdfsConfig = new HdfsConfig();
-        HdfsConfiguration hdfsConfiguration = new HiveHdfsConfiguration(new HdfsConfigurationInitializer(hdfsConfig), ImmutableSet.of());
-        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(hdfsConfiguration, hdfsConfig, new NoHdfsAuthentication());
-
-        metastore = new FileHiveMetastore(
-                new NodeVersion("test_version"),
-                hdfsEnvironment,
-                new MetastoreConfig(),
-                new FileHiveMetastoreConfig()
-                        .setCatalogDirectory(baseDir.toURI().toString())
-                        .setMetastoreUser("test"));
-
-        TestingIcebergPlugin icebergPlugin = new TestingIcebergPlugin(metastore, false);
-
-        queryRunner.installPlugin(icebergPlugin);
-        queryRunner.createCatalog(CATALOG, "iceberg", ImmutableMap.of("iceberg.unique-table-location", "true"));
-        queryRunner.installPlugin(new TestingHivePlugin(metastore));
-        queryRunner.createCatalog("hive", "hive", ImmutableMap.of("hive.security", "sql-standard"));
-        return queryRunner;
-    }
-
-    @BeforeClass
-    public void setUp()
-    {
-        assertQuerySucceeds(format("CREATE SCHEMA %s.%s", CATALOG, SCHEMA));
-    }
-
-    @AfterMethod
-    public void clean()
-    {
-        query(format("DROP TABLE IF EXISTS %s.%s.%s", CATALOG, SCHEMA, TABLE));
-        query(format("DROP TABLE IF EXISTS %s.%s.%s", CATALOG, SCHEMA, TABLE_RENAMED));
+        File tempDir = Files.createTempDirectory("test_iceberg_v2").toFile();
+        metastoreDir = new File(tempDir, "iceberg_data");
```

Comment:
however, it seems the test doesn't need to know the `metastoreDir`, so instead of setting it up (and cleaning it later), it should be sufficient to pass `Optional.empty()` for the `metastoreDir`, right?

Reply:
we need `metastoreDir` because it is used in `createTestingFileHiveMetastore(metastoreDir)`, 
but I removed two levels nesting